src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java

```diff
@@ -61,40 +84,136 @@ protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOE
+                builder.setPathPrefix(settings.get(CLIENT_PATH_PREFIX));
+            }
+        }
+    }
+
     @Before
-    public void initPaClient() throws IOException {
+    public void initPaClient() throws Exception {
         String cluster = System.getProperty("tests.rest.cluster");
         logger.info("Cluster is {}", cluster);
         if (cluster == null) {
             throw new RuntimeException("Must specify [tests.rest.cluster] system property with a comma delimited list of [host:port] "
                     + "to which to send REST requests");
         }
         List<HttpHost> hosts = Collections.singletonList(
-                buildHttpHost(cluster.substring(0, cluster.lastIndexOf(":")), PORT));
+                new HttpHost(cluster.substring(0, cluster.lastIndexOf(":")), PORT, "http"));
         logger.info("initializing PerformanceAnalyzer client against {}", hosts);
-        paClient = buildClient(restClientSettings(), hosts.toArray(new HttpHost[0]));
+        paClient = buildBasicClient(restClientSettings(), hosts.toArray(new HttpHost[0]));
     }
 
+    private enum Component {
+        PA,
+        RCA
+    }
 
-    public static void ensurePaAndRcaEnabled() throws Exception {
-        // TODO replace with waitFor with a 1min timeout
-        for (int i = 0; i < 60; i++) {
-            Response resp = client().performRequest(new Request("GET", "_opendistro/_performanceanalyzer/cluster/config"));
-            Map<String, Object> respMap = mapper.readValue(EntityUtils.toString(resp.getEntity(), "UTF-8"),
-                    new TypeReference<Map<String, Object>>(){});
-            if (respMap.get("currentPerformanceAnalyzerClusterState").equals(3) &&
-                    !respMap.get("currentPerformanceAnalyzerClusterState").equals(7)) {
-                break;
-            }
-            Thread.sleep(1000L);
-        }
-        Response resp = client().performRequest(new Request("GET", "_opendistro/_performanceanalyzer/cluster/config"));
-        Map<String, Object> respMap = mapper.readValue(EntityUtils.toString(resp.getEntity(), "UTF-8"),
-                new TypeReference<Map<String, Object>>(){});
-        if (!respMap.get("currentPerformanceAnalyzerClusterState").equals(3) &&
-                !respMap.get("currentPerformanceAnalyzerClusterState").equals(7)) {
-            throw new Exception("PA and RCA are not enabled on the target cluster!");
+    /**
+     * enableComponent enables PA or RCA on the test cluster
+     * @param component Either PA or RCA
+     * @return The cluster's {@link Response}
+     */
+    public Response enableComponent(Component component) throws Exception {
+        String endpoint;
+        if (component == Component.PA) {
+            endpoint = "_opendistro/_performanceanalyzer/cluster/config";
+        } else if (component == Component.RCA) {
+            endpoint = "_opendistro/_performanceanalyzer/rca/cluster/config";
+        } else {
```

Comment:
Should we make it a switch case as the component list will grow ?

Reply:
Added switch case and updated documentation